packages/router/src/private-apis.js

```diff
@@ -2,11 +2,20 @@
  * Internal dependencies
  */
 import { useHistory, useLocation, RouterProvider } from './router';
+import {
+	isPreviewingTheme,
+	currentlyPreviewingTheme,
+} from './is-previewing-theme';
+import { default as Link, useLink } from './link';
 import { lock } from './lock-unlock';
 
 export const privateApis = {};
 lock( privateApis, {
 	useHistory,
 	useLocation,
 	RouterProvider,
+	isPreviewingTheme,
+	currentlyPreviewingTheme,
+	Link,
+	useLink,
```

Comment:
I would be ok with moving Link and useLink to the router package if needed but only if we can remove the dependency to isPreviewingTheme and currentlyPreviewingTheme

Reply:
Yeah, same thought. One way to do that is to make any consumer of `useLink`/`Link` to check for the presence of the `wp_theme_preview` parameter, but that leaks that feature everywhere and every page/consumer has to be aware of that. See what I know about theme previews at https://github.com/WordPress/gutenberg/pull/63849#discussion_r1687870181 Happy to hear alternatives.